Index: b/board/getinge/netcomv3/netcomv3.c
===================================================================
--- a/board/getinge/netcomv3/netcomv3.c	2012-11-26 16:00:41.606309540 +0100
+++ b/board/getinge/netcomv3/netcomv3.c	2012-11-26 16:02:15.679167052 +0100
@@ -168,8 +168,8 @@
 
 	erstl = readl(&rstc->mr) & AT91_RSTC_MR_ERSTL_MASK;
 
-	/* Need to reset PHY -> 500ms reset */
-	writel(AT91_RSTC_KEY | AT91_RSTC_MR_ERSTL(13) |
+	/* Need to reset PHY -> 15.6ms reset */
+	writel(AT91_RSTC_KEY | AT91_RSTC_MR_ERSTL(8) |
 		AT91_RSTC_MR_URSTEN, &rstc->mr);
 
 	writel(AT91_RSTC_KEY | AT91_RSTC_CR_EXTRST, &rstc->cr);
