--- inetutils-1.8.orig/src/Makefile.am	2010-05-15 19:25:47.000000000 +0800
+++ inetutils-1.8/src/Makefile.am	2011-01-31 17:46:48.218097964 +0800
@@ -96,8 +96,6 @@
 
 AM_CPPFLAGS = $(PATHDEF_BSHELL) $(PATHDEF_CONSOLE) $(PATHDEF_CP) $(PATHDEF_DEFPATH) $(PATHDEF_DEV) $(PATHDEF_INETDCONF) $(PATHDEF_INETDDIR) $(PATHDEF_INETDPID) $(PATHDEF_KLOG) $(PATHDEF_LOG) $(PATHDEF_LOGCONF) $(PATHDEF_LOGIN) $(PATHDEF_LOGPID) $(PATHDEF_NOLOGIN) $(PATHDEF_RLOGIN) $(PATHDEF_RSH) $(PATHDEF_TTY) $(PATHDEF_UTMP) $(PATHDEF_UUCICO)
 
-SUIDMODE = -o root -m 4775
-
 install-exec-hook: @traceroute_INSTALL_HOOK@ @rsh_INSTALL_HOOK@ @rlogin_INSTALL_HOOK@ @rcp_INSTALL_HOOK@
 
 install-traceroute-hook:
--- inetutils-1.8.orig/src/Makefile.in	2010-05-16 04:29:57.000000000 +0800
+++ inetutils-1.8/src/Makefile.in	2011-01-31 17:47:05.974098524 +0800
@@ -1078,7 +1078,6 @@
 	$(LIBAUTH) $(LIBCRYPT) $(LIBUTIL)
 
 AM_CPPFLAGS = $(PATHDEF_BSHELL) $(PATHDEF_CONSOLE) $(PATHDEF_CP) $(PATHDEF_DEFPATH) $(PATHDEF_DEV) $(PATHDEF_INETDCONF) $(PATHDEF_INETDDIR) $(PATHDEF_INETDPID) $(PATHDEF_KLOG) $(PATHDEF_LOG) $(PATHDEF_LOGCONF) $(PATHDEF_LOGIN) $(PATHDEF_LOGPID) $(PATHDEF_NOLOGIN) $(PATHDEF_RLOGIN) $(PATHDEF_RSH) $(PATHDEF_TTY) $(PATHDEF_UTMP) $(PATHDEF_UUCICO)
-SUIDMODE = -o root -m 4775
 all: all-am
 
 .SUFFIXES:
--- inetutils-1.8.orig/ping/Makefile.am	2010-05-15 19:25:47.000000000 +0800
+++ inetutils-1.8/ping/Makefile.am	2011-01-31 17:47:19.222098520 +0800
@@ -30,8 +30,6 @@
   ping_router.c ping_timestamp.c ping_common.h  ping_impl.h ping.h libping.c
 ping6_SOURCES = ping6.c ping_common.c ping_common.h ping6.h
 
-SUIDMODE = -o root -m 4775
-
 install-ping-hook:
 	-@for x in $(bin_PROGRAMS); do \
 	$(INSTALL_PROGRAM) $$x $(SUIDMODE) $(AM_INSTALL_PROGRAM_FLAGS) $(DESTDIR)$(bindir)/`echo $$x|sed '$(transform)'` ; \
--- inetutils-1.8.orig/ping/Makefile.in	2010-05-16 04:29:56.000000000 +0800
+++ inetutils-1.8/ping/Makefile.in	2011-01-31 17:47:26.294348128 +0800
@@ -976,7 +976,6 @@
   ping_router.c ping_timestamp.c ping_common.h  ping_impl.h ping.h libping.c
 
 ping6_SOURCES = ping6.c ping_common.c ping_common.h ping6.h
-SUIDMODE = -o root -m 4775
 all: all-am
 
 .SUFFIXES:
