--- a/lib/argp.h	2011-10-11 14:59:39.362050856 +0800
+++ b/lib/argp.h	2011-10-11 14:29:02.554050768 +0800
@@ -519,7 +519,7 @@
                                FILE *__restrict __stream,
                                unsigned int __flags);
 
-#if _LIBC || !defined __USE_EXTERN_INLINES
+#if _LIBC || !defined INET_USE_EXTERN_INLINES
 /* Possibly output the standard usage message for ARGP to stderr and exit.  */
 extern void argp_usage (const struct argp_state *__state);
 extern void __argp_usage (const struct argp_state *__state);
@@ -552,7 +552,7 @@
                             const char *__restrict __fmt, ...)
      __attribute__ ((__format__ (__printf__, 4, 5)));
 
-#if _LIBC || !defined __USE_EXTERN_INLINES
+#if _LIBC || !defined INET_USE_EXTERN_INLINES
 /* Returns true if the option OPT is a valid short option.  */
 extern int _option_is_short (const struct argp_option *__opt) __THROW;
 extern int __option_is_short (const struct argp_option *__opt) __THROW;
@@ -572,7 +572,7 @@
                            const struct argp_state *__restrict __state)
      __THROW;
 
-#ifdef __USE_EXTERN_INLINES
+#ifdef INET_USE_EXTERN_INLINES
 
 # if !_LIBC
 #  define __argp_usage argp_usage
--- a/lib/argp-xinl.c	2011-10-11 14:59:56.082050560 +0800
+++ b/lib/argp-xinl.c	2011-10-11 14:28:43.162051079 +0800
@@ -24,8 +24,8 @@
 # include <features.h>
 #endif
 
-#ifndef __USE_EXTERN_INLINES
-# define __USE_EXTERN_INLINES   1
+#ifndef INET_USE_EXTERN_INLINES
+# define INET_USE_EXTERN_INLINES   1
 #endif
 #define ARGP_EI
 #undef __OPTIMIZE__
