diff --git a/src/dep/msg.c b/src/dep/msg.c
index 31200c0..41d0a97 100644
--- a/src/dep/msg.c
+++ b/src/dep/msg.c
@@ -156,7 +156,9 @@ msgPackAnnounce(void *buf, PtpClock * ptpClock)
 		flip16(ptpClock->clockQuality.offsetScaledLogVariance);
 	*(UInteger8 *) (buf + 52) = ptpClock->grandmasterPriority2;
 	memcpy((buf + 53), ptpClock->grandmasterIdentity, CLOCK_IDENTITY_LENGTH);
-	*(UInteger16 *) (buf + 61) = flip16(ptpClock->stepsRemoved);
+	/* avoid misaligned word write at odd address */
+	*(UInteger8 *) (buf + 61) = (UInteger8)(ptpClock->stepsRemoved >> 8);
+	*(UInteger8 *) (buf + 62) = (UInteger8)ptpClock->stepsRemoved;
 	*(Enumeration8 *) (buf + 63) = ptpClock->timeSource;
 }
 
@@ -181,7 +183,9 @@ msgUnpackAnnounce(void *buf, MsgAnnounce * announce)
 	announce->grandmasterPriority2 = *(UInteger8 *) (buf + 52);
 	memcpy(announce->grandmasterIdentity, (buf + 53), 
 	       CLOCK_IDENTITY_LENGTH);
-	announce->stepsRemoved = flip16(*(UInteger16 *) (buf + 61));
+	/* avoid misaligned word read at odd address */
+	announce->stepsRemoved = *(UInteger8 *) (buf + 61);
+	announce->stepsRemoved = (announce->stepsRemoved << 8) | *(UInteger8 *) (buf + 62);
 	announce->timeSource = *(Enumeration8 *) (buf + 63);
 	
 #ifdef PTPD_DBG
diff --git a/src/dep/net.c b/src/dep/net.c
index a84b62f..3bc09df 100644
--- a/src/dep/net.c
+++ b/src/dep/net.c
@@ -394,9 +394,11 @@ again:
 		}
 	}
 
-	if (!ts || !matched) {
+	if (!matched)
+		goto again;
+
+	if (!ts)
 		return -1;
-	}
 
 	/* Push packet onto stack. */
 	index = netPath->tx_stack.count;
