diff --git a/drivers/usb/musb/tusb6010_omap.c b/drivers/usb/musb/tusb6010_omap.c
index c784e6c..9107438 100644
--- a/drivers/usb/musb/tusb6010_omap.c
+++ b/drivers/usb/musb/tusb6010_omap.c
@@ -87,6 +87,7 @@ static int tusb_omap_dma_stop(struct dma_controller *c)
 static inline int tusb_omap_use_shared_dmareq(struct tusb_omap_dma_ch *chdat)
 {
 	u32		reg = musb_readl(chdat->tbase, TUSB_DMA_EP_MAP);
+	struct musb		*musb = chdat->musb;
 
 	if (reg != 0) {
 		dev_dbg(musb->controller, "ep%i dmareq0 is busy for ep%i\n",
