diff --git a/arch/arm/kernel/vmlinux.lds.S b/arch/arm/kernel/vmlinux.lds.S
index e5287f2..94d35e3 100644
--- a/arch/arm/kernel/vmlinux.lds.S
+++ b/arch/arm/kernel/vmlinux.lds.S
@@ -73,7 +73,7 @@ SECTIONS
 		INIT_CALLS
 		CON_INITCALL
 		SECURITY_INITCALL
-		INIT_RAM_FS
+		/*INIT_RAM_FS*/
 
 #ifndef CONFIG_XIP_KERNEL
 		__init_begin = _stext;
@@ -151,6 +151,10 @@ SECTIONS
 #endif
 
 	_etext = .;			/* End of text and rodata section */
+	.ramfs : {
+		INIT_RAM_FS
+	}
+
 
 #ifdef CONFIG_XIP_KERNEL
 	__data_loc = ALIGN(4);		/* location in binary */
