--- alsa-lib.old/include/local.h	2012-07-27 18:51:37.578103769 +0800
+++ alsa-lib/include/local.h	2012-07-27 18:52:13.030103495 +0800
@@ -281,4 +281,6 @@
                                   const char *base, const char *key,
 				  snd_config_t **result);
 
+int versionsort(const void *a, const void *b);
+
 #endif
--- alsa-lib.old/src/ucm/parser.c	2012-07-27 18:52:23.834352541 +0800
+++ alsa-lib/src/ucm/parser.c	2012-07-27 18:52:29.626103057 +0800
@@ -1321,7 +1321,7 @@
 	}
 }
 
-static int versionsort(const void *a, const void *b)
+int versionsort(const void *a, const void *b)
 {
 	return __strverscmp ((*(const struct dirent **) a)->d_name,
 				(*(const struct dirent **) b)->d_name);
