--- alsa-lib-1.0.24.1/src/conf/cards/Makefile.am.org	2012-03-30 16:26:21.595409660 +0800
+++ alsa-lib-1.0.24.1/src/conf/cards/Makefile.am	2012-03-30 16:05:00.371408880 +0800
@@ -2,6 +2,12 @@
 alsadir = $(alsaconfigdir)/cards
 cfg_files = aliases.conf \
 	AACI.conf \
+	AD1980.conf \
+	AD1980-no-mmap.conf \
+	AD73322.asound.conf \
+	AD73322.conf \
+	ADAU1371.conf \
+	ADAU1373.conf \
 	ATIIXP.conf \
 	ATIIXP-SPDMA.conf \
 	ATIIXP-MODEM.conf \
@@ -12,6 +18,10 @@
 	Audigy2.conf \
 	Aureon51.conf \
 	Aureon71.conf \
+	bfin-ad1836.conf \
+	bfin-ad193x.conf \
+	bfin-ad73311.conf \
+	bfin-ssm2602.conf \
 	CA0106.conf \
 	CMI8338.conf \
 	CMI8338-SWIEC.conf \
@@ -24,6 +34,7 @@
 	ENS1370.conf \
 	ENS1371.conf \
 	ES1968.conf \
+	eval-adau1x61.conf \
 	FM801.conf \
 	GUS.conf \
 	HDA-Intel.conf \
--- alsa-lib-1.0.26/src/conf/cards/Makefile.in.bak	2013-03-19 17:16:28.562091381 +0800
+++ alsa-lib-1.0.26/src/conf/cards/Makefile.in	2013-03-19 17:18:56.491446996 +0800
@@ -226,6 +226,9 @@
 	SI7018.conf SB-XFi.conf TRID4DWAVENX.conf USB-Audio.conf \
 	YMF744.conf VIA686A.conf VIA8233.conf VIA8233A.conf \
 	VIA8237.conf VX222.conf VXPocket.conf VXPocket440.conf \
+	AD1980.conf AD1980-no-mmap.conf AD73322.asound.conf \
+	AD73322.conf ADAU1371.conf ADAU1373.conf eval-adau1x61.conf \
+	bfin-ad1836.conf bfin-ad193x.conf bfin-ad73311.conf bfin-ssm2602.conf \
 	$(am__append_1)
 alsa_DATA = $(cfg_files)
 @BUILD_ALISP_TRUE@SI7018dir = $(alsaconfigdir)/cards/SI7018
