Index: u-boot-1.3.4/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c
===================================================================
--- u-boot-1.3.4.orig/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c	2011-08-01 15:41:01.376076713 +0200
+++ u-boot-1.3.4/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c	2011-08-01 15:41:41.216076735 +0200
@@ -124,6 +124,13 @@
 }
 #endif
 
+#if defined(CONFIG_CMD_USB)
+static void at91samm10g45ek_usb_hw_init(void)
+{
+       at91_set_gpio_output(AT91_PIN_PB28, 1);
+}
+#endif
+
 #ifdef CONFIG_MACB
 static void at91samm10g45ek_macb_hw_init(void)
 {
@@ -320,6 +327,9 @@
 #ifdef CONFIG_LCD
 	at91samm10g45ek_lcd_hw_init();
 #endif
+#ifdef CONFIG_CMD_USB
+	at91samm10g45ek_usb_hw_init();
+#endif
 	return 0;
 }
 
