Index: u-boot-1.3.4/include/configs/at91sam9m10g45ek.h
===================================================================
--- u-boot-1.3.4.orig/include/configs/at91sam9m10g45ek.h	2011-09-06 16:45:51.628104839 +0200
+++ u-boot-1.3.4/include/configs/at91sam9m10g45ek.h	2011-09-06 16:46:17.048104826 +0200
@@ -198,6 +198,8 @@
 #define CONFIG_BAUDRATE		115200
 #define CFG_BAUDRATE_TABLE	{115200 , 19200, 38400, 57600, 9600 }
 
+#define CFG_HUSH_PARSER
+#define CFG_PROMPT_HUSH_PS2     "> "
 #define CFG_PROMPT		"U-Boot> "
 #define CFG_CBSIZE		256
 #define CFG_MAXARGS		16
