Index: u-boot-1.3.4/include/asm-arm/mach-types.h
===================================================================
--- u-boot-1.3.4.orig/include/asm-arm/mach-types.h	2011-08-01 15:42:34.546076763 +0200
+++ u-boot-1.3.4/include/asm-arm/mach-types.h	2011-08-01 15:44:09.066076815 +0200
@@ -1602,6 +1602,7 @@
 #define MACH_TYPE_AT91SAM9G45EKES      2212
 #define MACH_TYPE_AT91SAM9G20EK_2MMC   2288
 #define MACH_TYPE_AT91SAM9M10EKES      2509
+#define MACH_TYPE_GETINGE_NETCOMV3     3602
 
 #ifdef CONFIG_ARCH_EBSA110
 # ifdef machine_arch_type
@@ -20684,7 +20685,17 @@
 # define machine_is_at91sam9m10ekes()	(0)
 #endif
 
-
+#ifdef CONFIG_MACH_GETINGE_NETCOMV3
+# ifdef machine_arch_type
+#  undef machine_arch_type
+#  define machine_arch_type	__machine_arch_type
+# else
+#  define machine_arch_type	MACH_TYPE_GETINGE_NETCOMV3
+# endif
+# define machine_is_getinge_netcomv3()	(machine_arch_type == MACH_TYPE_GETINGE_NETCOMV3)
+#else
+# define machine_is_getinge_netcomv3()	(0)
+#endif
 
 /*
  * These have not yet been registered
Index: u-boot-1.3.4/Makefile
===================================================================
--- u-boot-1.3.4.orig/Makefile	2011-08-01 15:45:50.336076871 +0200
+++ u-boot-1.3.4/Makefile	2011-08-01 15:46:51.036076903 +0200
@@ -2514,6 +2514,22 @@
 	fi;
 	@$(MKCONFIG) -a at91sam9m10g45ek arm arm926ejs at91sam9m10g45ek atmel at91sam9
 
+getinge_netcomv3_nandflash_config \
+getinge_netcomv3_dataflash_config \
+getinge_netcomv3_dataflash_cs0_config \
+getinge_netcomv3_config	:	unconfig
+	@mkdir -p $(obj)include
+	@if [ "$(findstring _nandflash,$@)" ] ; then \
+		echo "#define CFG_USE_NANDFLASH 1"	>>$(obj)include/config.h ; \
+		$(XECHO) "... with environment variable in NAND FLASH" ; \
+		echo "#define CFG_USE_GETINGE_NETCOMV3 1"	>>$(obj)include/config.h ; \
+	else \
+		echo "#define CFG_USE_DATAFLASH 1"	>>$(obj)include/config.h ; \
+		$(XECHO) "... with environment variable in SPI DATAFLASH CS0" ; \
+		echo "#define CFG_USE_GETINGE_NETCOMV3 1"	>>$(obj)include/config.h ; \
+	fi;
+	@$(MKCONFIG) -a at91sam9m10g45ek arm arm926ejs at91sam9m10g45ek atmel at91sam9
+
 at91sam9g45ekes_nandflash_config \
 at91sam9g45ekes_dataflash_config \
 at91sam9g45ekes_dataflash_cs0_config \
Index: u-boot-1.3.4/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c
===================================================================
--- u-boot-1.3.4.orig/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c	2011-08-01 15:44:20.076076821 +0200
+++ u-boot-1.3.4/board/atmel/at91sam9m10g45ek/at91sam9m10g45ek.c	2011-08-01 15:44:49.516076837 +0200
@@ -309,6 +309,8 @@
 	gd->bd->bi_arch_number = MACH_TYPE_AT91SAM9G45EKES;
 #elif defined CONFIG_AT91SAM9M10EKES
 	gd->bd->bi_arch_number = MACH_TYPE_AT91SAM9M10EKES;
+#elif defined CONFIG_GETINGE_NETCOMV3
+	gd->bd->bi_arch_number = MACH_TYPE_GETINGE_NETCOMV3;
 #endif
 	/* adress of boot parameters */
 	gd->bd->bi_boot_params = PHYS_SDRAM + 0x100;
Index: u-boot-1.3.4/include/configs/at91sam9m10g45ek.h
===================================================================
--- u-boot-1.3.4.orig/include/configs/at91sam9m10g45ek.h	2011-08-01 15:45:02.506076845 +0200
+++ u-boot-1.3.4/include/configs/at91sam9m10g45ek.h	2011-08-01 15:45:43.546076867 +0200
@@ -50,6 +50,10 @@
 #define CONFIG_AT91SAM9M10EKES  1       /* on an AT91SAM9M10EKES Board  */
 #endif
 
+#ifdef CFG_USE_GETINGE_NETCOMV3
+#define CONFIG_GETINGE_NETCOMV3 1
+#endif
+
 #undef CONFIG_USE_IRQ			/* we don't need IRQ/FIQ stuff	*/
 
 #define CONFIG_CMDLINE_TAG	1	/* enable passing of ATAGs	*/
