Index: u-boot-1.3.4/localversion-getinge
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ u-boot-1.3.4/localversion-getinge	2011-08-02 15:01:10.509369904 +0200
@@ -0,0 +1 @@
+-getinge-netcom
\ No newline at end of file
Index: u-boot-1.3.4/tools/setlocalversion
===================================================================
--- u-boot-1.3.4.orig/tools/setlocalversion	2011-08-02 14:58:37.029369982 +0200
+++ u-boot-1.3.4/tools/setlocalversion	2011-08-02 15:00:56.999369911 +0200
@@ -9,7 +9,7 @@
 cd "${1:-.}" || usage
 
 # Check for git and a git repo.
-if head=`git rev-parse --verify HEAD 2>/dev/null`; then
+if test -d .git && head=`git rev-parse --verify HEAD 2>/dev/null`; then
 	# Do we have an untagged version?
 	if git name-rev --tags HEAD | \
 	   grep -E '^HEAD[[:space:]]+(.*~[0-9]*|undefined)$' > /dev/null; then
@@ -30,7 +30,7 @@
 fi
 
 # Check for svn and a svn repo.
-if rev=`svn info 2>/dev/null` ; then
+if test -d .svn && rev=`svn info 2>/dev/null` ; then
 	rev=`echo "${rev}" | grep '^Revision' | awk '{print $NF}'`
 	printf -- '-svn%s' $rev
 fi
