Index: u-boot-1.3.4/lib_arm/eabi_compat.c
===================================================================
--- u-boot-1.3.4.orig/lib_arm/eabi_compat.c	2011-08-01 15:47:39.776076929 +0200
+++ u-boot-1.3.4/lib_arm/eabi_compat.c	2011-08-01 15:47:47.806076933 +0200
@@ -16,3 +16,8 @@
 	printf("raise: Signal # %d caught\n", signum);
 	return 0;
 }
+
+/* Dummy function to avoid linker complaints */
+void __aeabi_unwind_cpp_pr0(void)
+{
+}
