diff -Naur u-boot-1.3.4.orig/arm_config.mk u-boot-1.3.4/arm_config.mk
--- u-boot-1.3.4.orig/arm_config.mk	2010-04-21 16:09:47.000000000 -0400
+++ u-boot-1.3.4/arm_config.mk	2010-04-21 16:19:25.000000000 -0400
@@ -22,3 +22,32 @@
 #
 
 PLATFORM_CPPFLAGS += -DCONFIG_ARM -D__ARM__
+
+# Explicitly specifiy 32-bit ARM ISA since toolchain default can be -mthumb:
+PLATFORM_CPPFLAGS += $(call cc-option,-marm,)
+
+# Try if EABI is supported, else fall back to old API,
+# i. e. for example:
+# - with ELDK 4.2 (EABI supported), use:
+#	-mabi=aapcs-linux -mno-thumb-interwork
+# - with ELDK 4.1 (gcc 4.x, no EABI), use:
+#	-mabi=apcs-gnu -mno-thumb-interwork
+# - with ELDK 3.1 (gcc 3.x), use:
+#	-mapcs-32 -mno-thumb-interwork
+PLATFORM_CPPFLAGS += $(call cc-option,\
+				-mabi=aapcs-linux -mno-thumb-interwork,\
+				$(call cc-option,\
+					-mapcs-32,\
+					$(call cc-option,\
+						-mabi=apcs-gnu,\
+					)\
+				) $(call cc-option,-mno-thumb-interwork,)\
+			)
+
+# For EABI, make sure to provide raise()
+ifneq (,$(findstring -mabi=aapcs-linux,$(PLATFORM_CPPFLAGS)))
+# This file is parsed several times; make sure to add only once.
+ifeq (,$(findstring lib_arm/eabi_compat.o,$(PLATFORM_LIBS)))
+PLATFORM_LIBS += $(OBJTREE)/lib_arm/eabi_compat.o
+endif
+endif
\ No newline at end of file
diff -Naur u-boot-1.3.4.orig/config.mk u-boot-1.3.4/config.mk
--- u-boot-1.3.4.orig/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/config.mk	2010-04-21 16:19:25.000000000 -0400
@@ -115,7 +115,7 @@
 sinclude $(TOPDIR)/$(ARCH)_config.mk	# include architecture dependend rules
 endif
 ifdef	CPU
-sinclude $(TOPDIR)/cpu/$(CPU)/config.mk	# include  CPU	specific rules
+sinclude $(TOPDIR)/cpu/$(CPU)/config.mk		# include  CPU	specific rules
 endif
 ifdef	SOC
 sinclude $(TOPDIR)/cpu/$(CPU)/$(SOC)/config.mk	# include  SoC	specific rules
diff -Naur u-boot-1.3.4.orig/cpu/arm1136/config.mk u-boot-1.3.4/cpu/arm1136/config.mk
--- u-boot-1.3.4.orig/cpu/arm1136/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/arm1136/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -30,6 +30,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
-PLATFORM_CPPFLAGS +=$(call cc-option,-mno-thumb-interwork,)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/arm720t/config.mk u-boot-1.3.4/cpu/arm720t/config.mk
--- u-boot-1.3.4.orig/cpu/arm720t/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/arm720t/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -31,5 +31,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/arm920t/config.mk u-boot-1.3.4/cpu/arm920t/config.mk
--- u-boot-1.3.4.orig/cpu/arm920t/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/arm920t/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -30,5 +30,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/arm925t/config.mk u-boot-1.3.4/cpu/arm925t/config.mk
--- u-boot-1.3.4.orig/cpu/arm925t/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/arm925t/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -30,5 +30,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/arm926ejs/config.mk u-boot-1.3.4/cpu/arm926ejs/config.mk
--- u-boot-1.3.4.orig/cpu/arm926ejs/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/arm926ejs/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -30,5 +30,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/ixp/config.mk u-boot-1.3.4/cpu/ixp/config.mk
--- u-boot-1.3.4.orig/cpu/ixp/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/ixp/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -33,5 +33,4 @@
 # Supply options according to compiler version
 #
 # =========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/lh7a40x/config.mk u-boot-1.3.4/cpu/lh7a40x/config.mk
--- u-boot-1.3.4.orig/cpu/lh7a40x/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/lh7a40x/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -30,5 +30,4 @@
 # Supply options according to compiler version
 #
 # ========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/pxa/config.mk u-boot-1.3.4/cpu/pxa/config.mk
--- u-boot-1.3.4.orig/cpu/pxa/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/pxa/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -31,5 +31,4 @@
 # Supply options according to compiler version
 #
 # ========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/s3c44b0/config.mk u-boot-1.3.4/cpu/s3c44b0/config.mk
--- u-boot-1.3.4.orig/cpu/s3c44b0/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/s3c44b0/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -31,5 +31,4 @@
 # Supply options according to compiler version
 #
 # ========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/cpu/sa1100/config.mk u-boot-1.3.4/cpu/sa1100/config.mk
--- u-boot-1.3.4.orig/cpu/sa1100/config.mk	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/cpu/sa1100/config.mk	2010-04-21 16:19:51.000000000 -0400
@@ -31,5 +31,4 @@
 # Supply options according to compiler version
 #
 # ========================================================================
-PLATFORM_CPPFLAGS +=$(call cc-option,-mapcs-32,-mabi=apcs-gnu)
 PLATFORM_RELFLAGS +=$(call cc-option,-mshort-load-bytes,$(call cc-option,-malignment-traps,))
diff -Naur u-boot-1.3.4.orig/lib_arm/eabi_compat.c u-boot-1.3.4/lib_arm/eabi_compat.c
--- u-boot-1.3.4.orig/lib_arm/eabi_compat.c	1969-12-31 19:00:00.000000000 -0500
+++ u-boot-1.3.4/lib_arm/eabi_compat.c	2010-04-21 16:19:51.000000000 -0400
@@ -0,0 +1,18 @@
+/*
+ * Utility functions needed for (some) EABI conformant tool chains.
+ *
+ * (C) Copyright 2009 Wolfgang Denk <wd@denx.de>
+ *
+ * This program is Free Software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ */
+
+#include <common.h>
+
+int raise (int signum)
+{
+	printf("raise: Signal # %d caught\n", signum);
+	return 0;
+}
diff -Naur u-boot-1.3.4.orig/lib_arm/Makefile u-boot-1.3.4/lib_arm/Makefile
--- u-boot-1.3.4.orig/lib_arm/Makefile	2010-04-21 16:09:46.000000000 -0400
+++ u-boot-1.3.4/lib_arm/Makefile	2010-04-21 16:19:51.000000000 -0400
@@ -40,6 +40,17 @@
 
 SRCS	:= $(SOBJS-y:.o=.S) $(COBJS-y:.o=.c)
 OBJS	:= $(addprefix $(obj),$(SOBJS-y) $(COBJS-y))
+# Always build libarm.a
+TARGETS	:= $(LIB)
+
+# Build private libgcc only when asked for
+
+# For EABI conformant tool chains, provide eabi_compat()
+ifneq (,$(findstring -mabi=aapcs-linux,$(PLATFORM_CPPFLAGS)))
+TARGETS	+= $(obj)eabi_compat.o
+endif
+
+all:	$(TARGETS)
 
 $(LIB):	$(obj).depend $(OBJS)
 	$(AR) $(ARFLAGS) $@ $(OBJS)
