--- libiconv-1.12/configure.old	2012-07-04 15:17:49.446104050 +0800
+++ libiconv-1.12/configure	2012-07-04 15:19:13.846352668 +0800
@@ -29760,6 +29760,7 @@
   solaris*) OS="solaris" ;;
   osf*)     OS="osf" ;;
 esac
+if test "$enable_shared" = yes; then
 if test -n "$OS"; then
 
 
@@ -29769,7 +29770,6 @@
 fi
 
 
-if test "$enable_shared" = yes; then
   case "$host_os" in
     mingw* | cygwin*) is_woe32dll=yes ;;
     *) is_woe32dll=no ;;
