--- gettext-0.16.1/gettext-tools/configure.old	2012-07-03 18:35:01.938105694 +0800
+++ gettext-0.16.1/gettext-tools/configure	2012-07-03 18:36:10.858352436 +0800
@@ -50083,6 +50083,8 @@
 case "$host_os" in
   linux*)
 
+if test $enable_shared = yes; then
+
 cat >>confdefs.h <<\_ACEOF
 #define DYNLOAD_LIBEXPAT 1
 _ACEOF
@@ -50090,7 +50092,7 @@
     LIBEXPAT="-ldl"
     LTLIBEXPAT="-ldl"
 
-
+fi
     ;;
   *)
 
