diff --git a/source/lib/replace/snprintf.c b/source/lib/replace/snprintf.c
index c54d721..9781518 100644
--- a/source/lib/replace/snprintf.c
+++ b/source/lib/replace/snprintf.c
@@ -1231,7 +1231,7 @@ static int add_cnk_list_entry(struct pr_chunk_x **list,
 	return ret;
 }
 #endif
-
+#ifndef __NOMMU__
 #ifndef HAVE_C99_VSNPRINTF
  int fprintf(FILE *stream, const char *fmt, ...)
 {
@@ -1252,6 +1252,7 @@ static int add_cnk_list_entry(struct pr_chunk_x **list,
 	return ret;
 }
 #endif
+#endif
 
 #endif 
 
