--- libeXosip2-3.6.0.old/tools/sip_reg.c	2012-08-07 13:33:38.906103421 +0800
+++ libeXosip2-3.6.0/tools/sip_reg.c	2012-08-07 13:34:58.477786574 +0800
@@ -259,10 +259,18 @@
 	}
 #ifndef _WIN32_WCE
 	if (!nofork) {
+# ifdef __NOMMU__
+		int cpid = vfork ();
+# else
 		int cpid = fork();
+# endif
 
 		if (cpid)				/* parent */
+# ifdef __NOMMU__
+			_exit (0);
+# else
 			exit(0);
+# endif
 		/* child */
 		close(0);
 		close(1);
