--- bonnie-1.94/Makefile.in.old	2012-07-05 14:19:35.562103394 +0800
+++ bonnie-1.94/Makefile.in	2012-07-05 14:20:09.042352804 +0800
@@ -9,8 +9,8 @@
 #MORE_WARNINGS=-Weffc++ -Wcast-align
 WFLAGS=-Wall -W -Wshadow -Wpointer-arith -Wwrite-strings -pedantic -ffor-scope $(MORE_WARNINGS)
-CFLAGS=-O2 @debug@ -DNDEBUG $(WFLAGS) $(MORECFLAGS)
+CFLAGS+=-O2 @debug@ -DNDEBUG $(WFLAGS) $(MORECFLAGS)
 CXX=@CXX@ $(CFLAGS)
 LINK=@CXX@
 THREAD_LFLAGS=@thread_ldflags@
 
 INSTALL=@INSTALL@
@@ -34,7 +34,7 @@
 GETCHOBJS=$(GETCHSRC:.cpp=.o)
 
 bonnie++: $(BONOBJS)
-	$(LINK) -o bonnie++ $(BONOBJS) $(THREAD_LFLAGS)
+	$(LINK) -o bonnie++ $(BONOBJS) $(THREAD_LFLAGS) $(LDFLAGS)
 
 zcav: $(ZCAVOBJS)
 	$(LINK) -o zcav $(ZCAVOBJS) $(THREAD_LFLAGS)
--- bonnie-1.94/configure.old	2012-06-26 13:22:34.030103559 +0800
+++ bonnie-1.94/configure	2012-06-26 13:22:55.554103197 +0800
@@ -4314,8 +4314,6 @@
    linux_pthread="#define LINUX_PTHREAD"
 fi
 
-LDFLAGS=-lpthread
-
 cat >conftest.$ac_ext <<_ACEOF
 /* confdefs.h.  */
 _ACEOF
