diff --git a/Makefile.am b/Makefile.am
index 1c214c6..eca9449 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -441,21 +441,6 @@ endif
 
 unit_objects =
 
-if TEST
-unit_tests = unit/test-eir
-
-noinst_PROGRAMS += $(unit_tests)
-
-unit_test_eir_SOURCES = unit/test-eir.c src/eir.c src/glib-helper.c
-unit_test_eir_LDADD = lib/libbluetooth-private.la @GLIB_LIBS@ @CHECK_LIBS@
-unit_test_eir_CFLAGS = $(AM_CFLAGS) @CHECK_CFLAGS@
-unit_objects += $(unit_test_eir_OBJECTS)
-else
-unit_tests =
-endif
-
-TESTS = $(unit_tests)
-
 pkgconfigdir = $(libdir)/pkgconfig
 
 pkgconfig_DATA = bluez.pc
diff --git a/Makefile.tools b/Makefile.tools
index 5b1efb8..6caa595 100644
--- a/Makefile.tools
+++ b/Makefile.tools
@@ -160,9 +160,9 @@ endif
 if TEST
 sbin_PROGRAMS += test/hciemu
 
-bin_PROGRAMS += test/l2test test/rctest
+sbin_PROGRAMS += test/l2test test/rctest
 
-noinst_PROGRAMS += test/gaptest test/sdptest test/scotest \
+sbin_PROGRAMS += test/gaptest test/sdptest test/scotest \
 			test/attest test/hstest test/avtest test/ipctest \
 					test/lmptest test/bdaddr test/agent \
 					test/btiotest test/test-textfile \
diff --git a/acinclude.m4 b/acinclude.m4
index 1d6d736..36fbe85 100644
--- a/acinclude.m4
+++ b/acinclude.m4
@@ -387,7 +387,7 @@ AC_DEFUN([AC_ARG_BLUEZ], [
 	AM_CONDITIONAL(PAND, test "${pand_enable}" = "yes")
 	AM_CONDITIONAL(DUND, test "${dund_enable}" = "yes")
 	AM_CONDITIONAL(CUPS, test "${cups_enable}" = "yes")
-	AM_CONDITIONAL(TEST, test "${test_enable}" = "yes" && test "${check_found}" = "yes")
+	AM_CONDITIONAL(TEST, test "${test_enable}" = "yes")
 	AM_CONDITIONAL(TOOLS, test "${tools_enable}" = "yes")
 	AM_CONDITIONAL(BCCMD, test "${bccmd_enable}" = "yes")
 	AM_CONDITIONAL(PCMCIA, test "${pcmcia_enable}" = "yes")
