--- ncurses-5.7.orig/misc/Makefile.in	2007-03-31 23:54:17.000000000 +0800
+++ ncurses-5.7/misc/Makefile.in	2010-12-29 09:58:04.000000000 +0800
@@ -76,7 +76,7 @@
 
 sources :
 
-install : install.data install.libs
+install : install.libs
 
 install.data :	terminfo.tmp \
 		$(DESTDIR)$(libdir) \
@@ -121,7 +121,7 @@
 $(DESTDIR)$(tabsetdir) :
 	$(SHELL) $(top_srcdir)/mkdirs.sh $@
 
-uninstall : uninstall.data uninstall.libs
+uninstall : uninstall.libs
 
 uninstall.data :
 	-test -d $(DESTDIR)$(tabsetdir) && rm -rf $(DESTDIR)$(tabsetdir)
