--- dosfstools-3.0.12.old/Makefile	2013-06-12 17:30:49.000000000 +0800
+++ dosfstools-3.0.12/Makefile	2013-06-20 16:26:09.262341690 +0800
@@ -26,6 +26,7 @@
 SBINDIR = $(PREFIX)/sbin
 DOCDIR = $(PREFIX)/share/doc
 MANDIR = $(PREFIX)/share/man
+OBJS = boot.o check.o common.o fat.o file.o io.o lfn.o charconv.o
 
 #OPTFLAGS = -O2 -fomit-frame-pointer -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64
 OPTFLAGS = -O2 -fomit-frame-pointer -D_GNU_SOURCE $(shell getconf LFS_CFLAGS)
@@ -40,9 +41,11 @@
 
 build: fatlabel fsck.fat mkfs.fat
 
-fatlabel: boot.o check.o common.o fat.o file.o io.o lfn.o charconv.o fatlabel.o
+fatlabel: $(OBJS) fatlabel.o
+	$(CC) $(OBJS) fatlabel.o $(LDFLAGS) -L$(STAGING_DIR)/usr/lib -liconv -o fatlabel
 
-fsck.fat: boot.o check.o common.o fat.o file.o io.o lfn.o charconv.o fsck.fat.o
+fsck.fat: $(OBJS) fsck.fat.o
+	$(CC) $(OBJS) fsck.fat.o $(LDFLAGS) -L$(STAGING_DIR)/usr/lib -liconv -o fsck.fat
 
 mkfs.fat: mkfs.fat.o
 
