diff --git a/drivers/net/wireless/libertas/if_sdio.c b/drivers/net/wireless/libertas/if_sdio.c
index 8a3d301..08e3046 100644
--- a/drivers/net/wireless/libertas/if_sdio.c
+++ b/drivers/net/wireless/libertas/if_sdio.c
@@ -940,6 +940,13 @@ static int if_sdio_host_to_card(struct lbs_private *priv,
 	unsigned long flags;
 
 	lbs_deb_enter_args(LBS_DEB_SDIO, "type %d, bytes %d", type, nb);
+#ifdef CONFIG_BF54x
+	/*
+	 * Workaround for getting IP address fail through DHCP.
+	 * Because the data is sent too fast to sdio-wifi card.
+	 */
+	mdelay(20);
+#endif
 
 	card = priv->card;
 
