Index: zlib-1.2.5/gzwrite.c
===================================================================
--- zlib-1.2.5.orig/gzwrite.c	2011-12-13 04:12:48.949133998 -0500
+++ zlib-1.2.5/gzwrite.c	2011-12-13 04:13:46.089822000 -0500
@@ -3,6 +3,7 @@
  * For conditions of distribution and use, see copyright notice in zlib.h
  */
 
+#include "zutil.h"
 #include "gzguts.h"
 
 /* Local functions */
@@ -35,7 +36,7 @@
     strm->zfree = Z_NULL;
     strm->opaque = Z_NULL;
     ret = deflateInit2(strm, state->level, Z_DEFLATED,
-                       15 + 16, 8, state->strategy);
+                       MAX_WBITS + 16, DEF_MEM_LEVEL, state->strategy); //
     if (ret != Z_OK) {
         free(state->in);
         gz_error(state, Z_MEM_ERROR, "out of memory");
